aslam_cv_common/src/timer.cc

```diff
@@ -184,6 +185,44 @@ std::string Timing::SecondsToTimeString(double seconds) {
   return buffer;
 }
 
+void Timing::WriteToYamlFile(std::string const& path) {
+  const map_t& tag_map = Instance().tag_map_;
+
+  if (tag_map.empty()) {
+    return;
+  }
+
+  std::ofstream output_file(path);
+
+  if (!output_file) {
+    VLOG(0) << "Could not write timing: Unable to open file: " << path;
+    return;
+  }
+
+  VLOG(0) << "Writing timing to file: " << path;
+  for (const typename map_t::value_type& t : tag_map) {
+    const size_t i = t.second;
+
+    if (GetNumSamples(i) > 0) {
+      std::string label = t.first;
+
+      // We do not want colons or hashes in a label, as they might interfere
+      // with reading the yaml later.
+      std::replace(label.begin(), label.end(), ':', '_');
+      std::replace(label.begin(), label.end(), '#', '_');
+
+      output_file << label << ":" << std::endl;
```

Comment:
use `\n` instead of `std::endl` here and below as the latter will also flush the stream and will therefore be slower